aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -310,6 +319,16 @@ Tensor& mvlgamma_(Tensor& self, int64_t p) {
     return result;                                                     \
   }
 
+// IMPLEMENT_UNARY_FLOATING_OP_* macros are used for ops which support 
+// implicit type promotion (integral and boolean inputs to default floating type)
+// Check PR #33322 for more details on this
+#define IMPLEMENT_UNARY_FLOATING_OP_CORE(op)                           \
+  Tensor op(const Tensor& self) {                                      \
+    Tensor result = isIntegralType(self.scalar_type(), /*include_bool=*/ true) ? at::empty({0}, self.options().dtype(typeMetaToScalarType(c10::get_default_dtype()))) : at::empty({0}, self.options()); \
+    at::op##_out(result, self.to(result.scalar_type()));               \
```

Comment:
Per above, is this cast needed?

Reply:
This should be resolved here: https://github.com/pytorch/pytorch/pull/33322/commits/c2ff01ba86bc13ce422c8fb49126aabf547b2b29.